semantics/src/main/kotlin/com/strumenta/kolasu/semantics/indexing/Index.kt

```diff
@@ -0,0 +1,70 @@
+package com.strumenta.kolasu.semantics.indexing
+
+/**
+ * Abstract storage component for external symbols.
+ **/
+interface Index {
+    /**
+     * Retrieve first value matching [predicate] from the index.
+     * @param predicate the predicate to match
+     * @return the associated value (exception if none)
+     **/
+    fun get(predicate: (Any) -> Boolean): Any {
```

Comment:
Should we use an interface representing Symbols instead of Any in this and successive methods?